src/main/java/org/candlepin/subscriptions/tally/admin/DataMigration.java

```diff
@@ -29,6 +29,16 @@
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.jdbc.support.rowset.SqlRowSet;
 
+/**
+ * Implement this interface for long-running database migrations that need to be triggered and run
+ * in the background. Once the migration is complete, remove the implementing class. For code
+ * archeologists, previous implementations were
+ *
+ * <ul>
+ *   <li>HardwareMeasurementMigration
+ *   <li>OrgIdToTallySnapshotMigration
+ * </ul>
+ */
 public abstract class DataMigration {
```

Comment:
Shouldn't this class be moved to swatch-database as well?

Reply:
I don't think so.  This class is for migrations that are performed as the result of an API call.  Moving this class to swatch-database would mean that swatch-database would have to be REST aware, remain resident instead of just being a job, and would just generally make swatch-database substantially more complex.  I can see doing that in the future, but I believe it's out of scope for this card.

Reply:
Ok, thanks for the clarification.